src/libopal.c

```diff
@@ -2852,9 +2852,36 @@ gen_obj (char *asm_fn, char *obj_fn)
   /// Assert object file name is not null
   assert(obj_fn);
 
-  // TODO
+    /// Create the NASM call with -g, -f and -o flags
+    logger(DEBUG, "Calling NASM to assemble object.");
+    char NASM_cmd[1024] = { 0 };
+    sprintf(NASM_cmd, "NASM -g -f elf64 -o %s %s", asm_fn, obj_fn);
+
+    /// Check if asm_fn can be read
+    logger(DEBUG, "access (%s, R_OK)", asm_fn);
+    if (access(asm_fn, R_OK))
+    {
+        _PASS;
 
-  logger(DEBUG, "=== END ===");
+        /// Check if obj_fn can be written to
+        logger(DEBUG, "access (%s, W_OK)", obj_fn);
+        if (access(obj_fn,  W_OK))
```

Comment:
access() returns zero on success. 
So the if condition will fail when it should actually succeed. :)
Compare with EXIT_SUCCESS instead.